iac-extensions/docker/src/main/java/org/sonar/iac/docker/checks/RetrieveRemoteResourcesCheck.java

```diff
@@ -0,0 +1,95 @@
+import org.sonar.iac.docker.checks.utils.CommandDetector;
+import org.sonar.iac.docker.checks.utils.command.SeparatedList;
+import org.sonar.iac.docker.symbols.ArgumentResolution;
+import org.sonar.iac.docker.tree.api.RunInstruction;
+
+import static org.sonar.iac.docker.checks.utils.StringPredicate.startsWithIgnoreQuotes;
+
+@Rule(key = "S7026")
+public class RetrieveRemoteResourcesCheck implements IacCheck {
+
+  private static final String MESSAGE = "Replace this invocation of %s with the ADD instruction.";
+
+  private static final Predicate<String> WGET_DOWNLOAD_FLAG_PREDICATE = startsWithIgnoreQuotes("-O", "--output-document");
+  private static final Predicate<String> URL_PREDICATE = startsWithIgnoreQuotes("http");
+
+  // wget -O /path/to/resource https://example.com/resource
+  private static final CommandDetector WGET_DOWNLOAD_FLAG_FIRST_DETECTOR = CommandDetector.builder()
+    .with("wget")
+    .withOptionalRepeating(WGET_DOWNLOAD_FLAG_PREDICATE.negate())
+    .with(WGET_DOWNLOAD_FLAG_PREDICATE)
+    .withOptionalRepeating(URL_PREDICATE.negate())
+    .with(URL_PREDICATE)
+    .build();
+
+  // wget https://example.com/resource -O /path/to/resource
+  private static final CommandDetector WGET_URL_FIRST_DETECTOR = CommandDetector.builder()
+    .with("wget")
+    .withOptionalRepeating(URL_PREDICATE.negate())
+    .with(URL_PREDICATE)
+    .withOptionalRepeating(WGET_DOWNLOAD_FLAG_PREDICATE.negate())
+    .with(WGET_DOWNLOAD_FLAG_PREDICATE)
+    .build();
+
+  private static final List<CommandDetector> WGET_DETECTORS = List.of(WGET_DOWNLOAD_FLAG_FIRST_DETECTOR, WGET_URL_FIRST_DETECTOR);
+  private static final List<String> WGET_AUTH_FLAGS = List.of("--http-user", "--http-password", "--proxy-user", "--proxy-password", "--load-cookies");
+
+  @Override
+  public void initialize(InitContext init) {
+    init.register(RunInstruction.class, RetrieveRemoteResourcesCheck::check);
+  }
+
+  private static void check(CheckContext ctx, RunInstruction runInstruction) {
+    List<ArgumentResolution> resolvedArgument = CheckUtils.resolveInstructionArguments(runInstruction);
+
+    WGET_DETECTORS.forEach((CommandDetector detector) -> {
+      SeparatedList<List<ArgumentResolution>, String> splitCommands = ArgumentResolutionSplitter.splitCommands(resolvedArgument);
+      splitCommands.elements().forEach(args -> checkArgument(ctx, detector, args));
+    });
+  }
+
+  private static void checkArgument(CheckContext ctx, CommandDetector detector, List<ArgumentResolution> args) {
+    detector.search(args).forEach((CommandDetector.Command command) -> {
+      if (!containsWgetAuthenticationFlags(args)) {
+        var textRangeList = args.stream().map(a -> a.argument().textRange())
+          .toList();
+        var textRange = TextRanges.merge(textRangeList);
+        ctx.reportIssue(textRange, MESSAGE.formatted("wget"));
+      }
+    });
+  }
+
+  private static boolean containsWgetAuthenticationFlags(List<ArgumentResolution> args) {
```

Comment:
Did you consider using `withAnyFlagExcept()` on the `CommandDetector`? Not saying you should do it, because I'm not sure how easy it is to integrate it in this case here.

Reply:
I tried but it is not possible. The flags may be in any order and exclude flags may be in different forms, e.g.:
`--proxy-password=password` or `--proxy-password password`.
So the logic is split into 2 parts: 
* detect required flags and URL 
* exclude auth flags